### PR TITLE
Event handler

### PR DESCRIPTION
**Pull request by joelrebel**
#### What does this PR do

- Moves Orchestrator event handlers into versioned package.
- Implements an Orchestrator event handler for Controller updates.
- Prefix EventType with category in published event.

### CODE REVIEW

File: pkg/api/v1/types/types.go [hunk 1]
```diff
@@ -51,20 +53,67 @@ type ConditionUpdate struct {
...
+
+	return nil
+}
+
 // MergeExisting when given an existing condition, validates the update based on existing values
 // and returns a condition that can be passed to the repository for update.
 //
 // The resourceVersion is not updated here and is left for the repository Store to update.
 //
 // This method makes sure that update does not overwrite existing data inadvertently.
```
Comment (DoctorVin) [inline pkg/api/v1/types/types.go:108]: why would we want to disregard the resource versions on a merge?

**Review by DoctorVin — COMMENTED** [review]
A couple of minor comments for discussion. Code looks fine.

File: pkg/api/v1/types/types.go [hunk 2]
```diff
@@ -51,20 +53,67 @@ type ConditionUpdate struct {
...
+//
+// Note:
+// The ResourceVersion attribute is not validated for updates through events,
+// this is because the controllers do not perform requests for the existing condition
+// since implementing a Request-Reply pattern on the NATS Jetstream is tideous and not recommended.
+//
+// The NATS Jetstream guarantees ordered delivery, as long as there is a single consumer of the event,
+// for now we're deploying a single orchestrator instance in each facility and so this check is not required.
+// In the case that we require multiple orchestrators in a facility the stream would need to be partitioned
+// to ensure ordered delivery.
```
[the remark this answers is not in the document]
Comment (joelrebel) [reply pkg/api/v1/types/types.go:84]: I was promised ordered delivery as a pull subscriber on the Jetstream - when asked around on the official NATS slack channel. 
I tested this out in the sandbox and the message seem to be ordered, would be worth we compare the Jetstream configurations and test scenarios we have setup?

```bash
for m in 1 2 3 4;
do 
  nats publish --creds=/root/nsc/nkeys/creds/KO/controllers/conditionorc.creds -s nats://nats:4222 com.hollow.sh.controllers.commands.servers.inventory.inventoryOutofband "hello $m";
done

08:37:00 Published 7 bytes to "com.hollow.sh.controllers.commands.servers.inventory.inventoryOutofband"
08:37:00 Published 7 bytes to "com.hollow.sh.controllers.commands.servers.inventory.inventoryOutofband"
08:37:00 Published 7 bytes to "com.hollow.sh.controllers.commands.servers.inventory.inventoryOutofband"
08:37:00 Published 7 bytes to "com.hollow.sh.controllers.commands.servers.inventory.inventoryOutofband"


for m in 1 2 3 4;
do
  nats consumer sub --creds=/root/nsc/nkeys/creds/KO/controllers/conditionorc.creds -s nats://nats:4222 controllers alloy;
done

[08:37:20] subj: com.hollow.sh.controllers.commands.servers.inventory.inventoryOutofband / tries: 1 / cons seq: 5 / str seq: 46 / pending: 3
hello 1
Acknowledged message

[08:37:20] subj: com.hollow.sh.controllers.commands.servers.inventory.inventoryOutofband / tries: 1 / cons seq: 6 / str seq: 47 / pending: 2
hello 2
Acknowledged message

[08:37:20] subj: com.hollow.sh.controllers.commands.servers.inventory.inventoryOutofband / tries: 1 / cons seq: 7 / str seq: 48 / pending: 1
hello 3
Acknowledged message

[08:37:20] subj: com.hollow.sh.controllers.commands.servers.inventory.inventoryOutofband / tries: 1 / cons seq: 8 / str seq: 49 / pending: 0
hello 4
Acknowledged message
```

[the remark this answers is not in the document]
Comment (DoctorVin) [reply pkg/api/v1/types/types.go:84]: Deliveries are ordered iff there are subscribers waiting for the deliveries. I have seen this qualification made in some NATS documentation, but I can't lay my hands on it at the moment.

That said, it's quite possible we have some minor difference in configuration. I think it's worth digging into to understand.

File: pkg/api/v1/types/types.go (hunk 1 above)
Comment (DoctorVin) [reply pkg/api/v1/types/types.go:108]: I see what you're getting at, but I wonder if maybe we need a better mechanism for managing or protecting against concurrent access.

**Review by DoctorVin — APPROVED** [review]
LGTM, let's get it out so that we can iterate on it.

**Pull request merged**